game/src/core/components/DrawComponent.java

```diff
@@ -341,147 +324,82 @@ public void animationMap(final Map<String, Animation> animationMap) {
-            storage.put(lastDir, list);
-          }
-        }
+  private static void getResourcesRecursively(final String path, final List<URL> foundList)
+      throws IOException, URISyntaxException {
+    List<URL> resources = getResources(path);
+    for (URL url : resources) {
+      Path file = Paths.get(url.toURI().normalize());
+      if (Files.isRegularFile(file)) {
+        foundList.add(url);
+      } else if (Files.isDirectory(file)) {
+        String relPath = path + "/" + file.getFileName();
+        getResourcesRecursively(relPath, foundList);
+      } else {
+        throw new RuntimeException("Neither file nor directory: " + file);
       }
     }
-
-    // sort the files in lexicographic order (like the most os)
-    // animations will be played in order
-    storage.values().forEach(x -> x.sort(Comparator.comparing(IPath::pathString)));
-    // create animations
-    storage.forEach(
-        (name, textureSet) -> animationMap.put(name, Animation.fromCollection(textureSet)));
-    jar.close();
   }
 
   /**
-   * Load animations if the game is running in the IDE (or over the shell).
+   * Loading animation assets.
    *
-   * @param path Path to the animations.
+   * <p>Checks if the game is running in a JAR or not and will execute the corresponding loading
+   * logic.
    */
-  private void loadAnimationsFromIDE(final IPath path) {
-    URL url = DrawComponent.class.getResource("/" + path.pathString());
-    if (url != null) {
-      try {
-        try {
-          File apps = new File(url.toURI());
-          animationMap =
-              Arrays.stream(Objects.requireNonNull(apps.listFiles()))
-                  .filter(File::isDirectory)
-                  .collect(Collectors.toMap(File::getName, DrawComponent::allFilesFromDirectory));
-        } catch (IllegalArgumentException e) {
-          LOGGER.log(
-              CustomLogLevel.ERROR, "Could not load animations from directory: " + url.toURI(), e);
-        }
-      } catch (URISyntaxException e) {
-        LOGGER.log(CustomLogLevel.ERROR, "Could not load animations from directory", e);
+  private void loadAnimationAssets(final IPath path) {
+    final Map<String, List<IPath>> subdirectoryMap = new HashMap<>();
+    try {
+      List<URL> resources = new ArrayList<>();
+      getResourcesRecursively(path.pathString(), resources);
+      if (resources.isEmpty()) {
+        throw new FileNotFoundException(path.toString());
+      }
+      for (URL url : resources) {
+        Path file = Paths.get(url.toURI().normalize());
+        subdirectoryMap
+            .computeIfAbsent(file.getParent().getFileName().toString(), k -> new ArrayList<>())
+            .add(new SimpleIPath(file.toString()));
       }
+    } catch (Exception e) {
+      System.err.println("File not found: " + path);
+      throw new RuntimeException(e);
     }
+    // A Map with sorted values (IPath lists) in natural string order (ascending)
+    animationMap =
```

Comment:
was ist mit dieser map? in meiner review-vorschau ist dies das letzte statement in der methode. kein return, kein nix. wozu wird diese map berechnet?

Reply:
@tgrothe ah ... sry! in der gh-vorschau war nicht zu erkennen, dass das ein attribut der klasse ist ... dann macht es natürlich sinn :)

Reply:
> @tgrothe ah ... sry! in der gh-vorschau war nicht zu erkennen, dass das ein attribut der klasse ist ... dann macht es natürlich sinn :)

das ärgert mich auch (immer), weil man dann an der stelle auch keine code suggestions hinzufügen kann...